docs/MAESTRO.md

```diff
@@ -0,0 +1,111 @@
+# Setting up Maestro for Edge
+
+
+[Maestro](https://maestro.mobile.dev/) is an [open-source](https://github.com/mobile-dev-inc/maestro/) mobile testing framework for end-to-end testing automation. It is used to automate testing of various flows within the Edge Wallet.
+
+
+## When to use this guide
+
+
+- Your machine is running macOS 13.x and isn't already set up to locally compile the [Edge App](https://github.com/EdgeApp/edge-react-gui).
+
+
+## When not to use this guide
+
+
+- You have previously set up your machine as a development environment. In this case, install Maestro directly via the [getting started documentation](https://maestro.mobile.dev/getting-started/maestro-cli).
+
+
+## Table of Contents
+
+
+- [Setting up Maestro for Edge](#setting-up-maestro-for-edge)
+  - [When to use this guide](#when-to-use-this-guide)
+  - [When not to use this guide](#when-not-to-use-this-guide)
+  - [Table of Contents](#table-of-contents)
+  - [Prerequisites](#prerequisites)
+  - [Installation (macOS)](#installation-macos)
+  - [Running Maestro tests](#running-maestro-tests)
+- [Creating Maestro tests with Maestro Studio](#creating-maestro-tests-with-maestro-studio)
+
+
+
+## Prerequisites
+
+- macOS 13.x (Both Apple Silicon and Intel are supported)
+
+- An Android device with [USB debugging enabled](https://developer.android.com/studio/debug/dev-options#enable) and connected via USB to your computer.
+
+
+## Installation (macOS)
+
+
+1. Open the Terminal app (Located in Applications/Utilities).
+2. Install Apple Command Line Tools by pasting the following command in your terminal:
+
+
+```bash
+xcode-select --install
+```
+
+
+This will open a dialog box. Click "Install", followed by "Agree" to the license terms. This will start the installation of the Apple Command Line Tools.
+
+
+> Note: Downloading and installing the Apple Command Line Tools can take between 10 minutes to an hour depending on your internet connection speed.
+
+
+3. Once Apple Command Line Tools has finished installing, run the following command to verify that the installation was successful:
+
+
+```bash
+xcode-select -p
+
+# /Library/Developer/CommandLineTools
+
+```
+
+4. Next, run the convenience shell script to install all dependencies required for Maestro, including Maestro itself:
+
+
+```bash
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/EdgeApp/edge-react-gui/maestro.sh)"
```

Comment:
Note for reviewers: this command will not work until this PR has been merged - until then the correct command is:

```bash
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/EdgeApp/edge-react-gui/itay/maestro/readme/maestro.sh)"
```